kubernetes/README.adoc

```diff
@@ -0,0 +1,78 @@
+= Kubernetes and Openshift tutorial
+
+This tutorial contains instructions on how to run Infinispan library mode (as a microservice) in Kubernetes/OpenShift.
+
+Prerequisites: Maven and Docker daemon running in the background.
+
+== Setting up OpenShift
+
+OpenShift Origin has a command line client called `oc` (it can be downloaded https://github.com/openshift/origin/releases[here]).
+
+The client allows you to spin up a local cluster (of course this is not recommended for production but for testing, it's perfect). In order to do that you need to invoke:
+
+```
+oc cluster up
+```
+
+== Setting up Kubernetes
+
+It is also possible to run this tutorial on Kubernetes. Installing a local cluster requires downloading and installing Minikube (detailed instruction might be found http://kubernetes.io/docs/getting-started-guides/minikube/[here]).
+
+Unfortunately Kubernetes installation does not use local Docker registry. The easiest way to overcome this problem is to push locally built tutorial into docker.io repository. The easiest way to do it is to use fabric8 plugin:
+
```

Comment:
This is a bit painful, having to push the image remotely and have minikube to pull it back on every change...
The trick to avoid it is to export the docker daemon from minikube. This is what is works for me:

1) Installed Minikube as usual: 
```
curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.8.0/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
```
2) Started minikube with ```minikube start --cpus=3 --memory=4096```
3) Executed ```eval $(minikube docker-env)```
4) mvn fabric8:run